pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3651,6 +3666,21 @@ func (c *Cloud) GetLoadBalancerName(ctx context.Context, clusterName string, ser
 	return cloudprovider.DefaultLoadBalancerName(service)
 }
 
+// GetAWSLoadBalancerName is a specific implementation of GetLoadBalancerName that could return an error
+func (c *Cloud) GetAWSLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) (string, error) {
```

Comment:
I think what you really want is the contents of `GetAWSLoadBalancerName` to be in `GetLoadBalancerName`

Reply:
and call `cloudprovider.DefaultLoadBalancerName(service)` if `annotationName == ""`.

Reply:
I'm a newbie in Go and just not sure if it is okay to break the `LoadBalancer interface` and return an error as the second variable from the function.

Reply:
@andrewsykim Do you think I need to update the interface to allow errors returning and add errors checking after each call of `GetLoadBalancerName` across all providers?:

 ```if err != nil {
		return err
	}```

Is it a good idea to do this refactoring inside this PR or it's better to create another one after this?

Reply:
If that is something that would need to be done, it would need to be in a new PR, @KIVagant, because a lot of files would have to be updated.